chain/client/src/gc_actor.rs

```diff
@@ -0,0 +1,102 @@
+use crate::metrics;
+use actix::{Actor, Addr, Arbiter, ArbiterHandle, AsyncContext, Context};
+use near_chain::{types::RuntimeAdapter, ChainStore, ChainStoreAccess};
+use near_chain_configs::{ClientConfig, GCConfig};
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::types::BlockHeight;
+use near_store::{metadata::DbKind, Store};
+use std::sync::Arc;
+use tracing::warn;
+
+pub struct GCActor {
+    store: ChainStore,
+    runtime_adapter: Arc<dyn RuntimeAdapter>,
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    gc_config: GCConfig,
+    is_archive: bool,
+}
+
+impl GCActor {
+    pub fn new(
+        store: Store,
+        genesis_height: BlockHeight,
+        runtime_adapter: Arc<dyn RuntimeAdapter>,
+        epoch_manager: Arc<dyn EpochManagerAdapter>,
+        gc_config: GCConfig,
+        is_archive: bool,
+    ) -> Self {
+        GCActor {
+            store: ChainStore::new(store, genesis_height, true),
+            runtime_adapter,
+            gc_config,
+            epoch_manager,
+            is_archive,
+        }
+    }
+
+    fn gc(&mut self, ctx: &mut Context<GCActor>) {
+        let timer = metrics::GC_TIME.start_timer();
+        if !self.is_archive {
```

Comment:
nit: I think this matches the deleted Chain::clear_data method, can you move the comments from there here?